core/src/main/java/io/grpc/internal/Http2ClientStreamTransportState.java

```diff
@@ -148,7 +148,9 @@ protected void transportDataReceived(ReadableBuffer frame, boolean endOfStream)
       if (endOfStream) {
         // This is a protocol violation as we expect to receive trailers.
         transportError =
-            Status.INTERNAL.withDescription("Received unexpected EOS on DATA frame from server.");
+            Status.INTERNAL.withDescription("Received unexpected EOS on DATA frame from server.")
+                .augmentDescription("DATA----------------------------\n"
+                    + ReadableBuffers.readAsString(frame, errorCharset));
```

Comment:
I was concerning it might be a large chunk of spam if the frame is very large. But the same pattern has been used lines above.